### PR TITLE
Navigation component

### PR DESCRIPTION
**Pull request by deioo**
Includes 2 in 1 responsive topbar and sidebar navigation.

http://cl.ly/1c1X2c3F1j47

### CODE REVIEW

File: components/_icon.scss [hunk 1]
```diff
@@ -1,4 +1,73 @@
...
+  globe: '\f026',
+  head: '\f027',
+  heart: '\f028',
+  image: '\f029',
+  left-arrow: '\f02a',
+  link: '\f02b',
+  lock: '\f02c',
+  mail: '\f02d',
+  menu: '\f02e',
+  'notification': '\f02f',
```
Comment (deioo) [inline components/_icon.scss]: Once we update libsass, we're going to have to remove this.
Libsass bug https://github.com/sass/libsass/issues/934

File: components/_navigation.scss [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+#{navigation-settings(sidebar, selector)} {
+  background: navigation-settings(sidebar, background-color);
+  bottom: 0;
+  color: navigation-settings(sidebar, text-color);
+  left: 0;
+  overflow: hidden;
+  position: fixed;
+  top: 0;
+  transform: translateX(navigation-settings(height-negative));
+  transition: transform navigation-settings(transition-duration), width navigation-settings(transition-duration);
```
Comment (deioo) [inline components/_navigation.scss]: Move on two lines

File: components/_navigation.scss [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+    }
+
+    .submenu li {
+      height: navigation-settings(height);
+    }
+  }
+}
+
+@media #{$small-only} {
+  .application {
```
Comment (deioo) [inline components/_navigation.scss]: Use `$layout-application-selector`

File: components/_navigation.scss [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+    border-bottom: 1px solid navigation-settings(sidebar, hover-color);
+    color: color(white);
+    font-size: $base-font-size;
+    line-height: navigation-settings(height);
+    list-style: none;
+    margin: 0;
+    padding: 0;
+    text-align: left;
+    text-transform: none;
+
```
Comment (carlosepp) [inline components/_navigation.scss]: extra line

File: components/_navigation.scss [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+    }
+
+    a#{navigation-settings(icon-class-selector)} {
+      @extend %paint-icon;
+
+      &:before {
+        width: navigation-settings(sidebar, min-width);
+      }
+    }
+
```
Comment (carlosepp) [inline components/_navigation.scss]: extra line

**Comment by carlosepp** [comment]
LGTM :+1:

**Pull request merged**